tx/transaction.go

```diff
@@ -24,11 +24,19 @@ import (
 
 var (
 	errIntrinsicGasOverflow = errors.New("intrinsic gas overflow")
+	ErrInvalidTxType        = errors.New("transaction type not valid in this context")
```

Comment:
what does this error mean ?

Reply:
It has a slightly different meaning from `ErrTxTypeNotSupported` but we don't actually need to make such distinction: I'll remove it